src/algebra/lie_algebras.lean

```diff
@@ -0,0 +1,146 @@
+  repeat { rw h },
+  simp,
+end
+
+end ring_commutator
+
+/--
+A Lie ring is an additive group with compatible product, known as the bracket, satisfying the
+Jacobi identity. The bracket is not associative unless it is identically zero.
+-/
+class lie_ring (L : Type v) [add_comm_group L] extends has_bracket L :=
+(add_left : ∀ (x y z : L), ［x + y, z］ = ［x, z］ + ［y, z］)
+(add_right : ∀ (x y z : L), ［z, x + y］ = ［z, x］ + ［z, y］)
+(alternate : ∀ (x : L), ［x, x］ = 0)
+(jacobi : ∀ (x y z : L), ［x, ［y, z］］ + ［y, ［z, x］］ + ［z, ［x, y］］ = 0)
+
+namespace lie_ring
+
+protected lemma skew (L : Type v) [add_comm_group L] [lie_ring L] (x y : L) :
+  ［x, y］ = -［y, x］ :=
+begin
+  rw [←sub_eq_zero_iff_eq, sub_neg_eq_add],
+  have h : ［x + y, x + y］ = 0 := by apply lie_ring.alternate,
+  rw [lie_ring.add_left,
+      lie_ring.add_right, lie_ring.add_right,
+      lie_ring.alternate, lie_ring.alternate] at h,
+  simp at h,
+  exact h,
+end
+
+instance of_associative_ring (A : Type v) [ring A] : lie_ring A :=
+{ add_left  := ring_commutator.add_left A,
+  add_right := ring_commutator.add_right A,
+  alternate := ring_commutator.alternate A,
+  jacobi    := ring_commutator.jacobi A }
+
+end lie_ring
+
+/--
+A Lie algebra is a module with compatible product, known as the bracket, satisfying the Jacobi
+identity. Forgetting the scalar multiplication, every Lie algebra is a Lie ring.
+-/
+class lie_algebra (R : Type u) (L : Type v)
+  [comm_ring R] [add_comm_group L] extends module R L, lie_ring L :=
+(smul_right : ∀ (t : R) (x y : L), ［x, t • y］ = t • ［x, y］)
+
+namespace lie_algebra
+
+protected lemma smul_left (R : Type u) (L : Type v)
+  [comm_ring R] [add_comm_group L] [lie_algebra R L] (t : R) (x y : L) :
+  ［t • x, y］ = t • ［x, y］ :=
+by { rw [lie_ring.skew, lie_ring.skew _ x y, lie_algebra.smul_right], simp }
+
+def adjoint_action (R : Type u) (L : Type v)
+  [comm_ring R] [add_comm_group L] [lie_algebra R L] (x : L) : L →ₗ[R] L :=
+{ to_fun := has_bracket.bracket x,
+  add    := by { intros, apply lie_ring.add_right },
+  smul   := by { intros, apply lie_algebra.smul_right } }
+
+def bracket_bilinear (R : Type u) (L : Type v)
+  [comm_ring R] [add_comm_group L] [lie_algebra R L] : L →ₗ[R] L →ₗ[R] L :=
+{ to_fun := lie_algebra.adjoint_action R L,
+  add    := by { unfold lie_algebra.adjoint_action, intros, ext, simp [lie_ring.add_left] },
+  smul   := by { unfold lie_algebra.adjoint_action, intros, ext, simp [lie_algebra.smul_left] } }
+
+instance of_associative_algebra (R : Type u) (A : Type v)
+  [comm_ring R] [ring A] [algebra R A] : lie_algebra R A :=
+begin
+  apply @lie_algebra.mk R A _ _ _ (lie_ring.of_associative_ring A),
+  intros t x y,
+  unfold has_bracket.bracket,
+  rw [algebra.mul_smul_comm, algebra.smul_mul_assoc, smul_sub],
+end
+
+instance of_endomorphism_algebra (R : Type u) (M : Type v)
+  [comm_ring R] [add_comm_group M] [module R M] : lie_algebra R (module.End R M) := infer_instance
+
+instance 𝖌𝖑 (n : Type u) (R : Type v)
```

Comment:
Additionally, I would try to avoid identifiers with unicode. Instances have a pretty standard naming scheme (`matrix.lie_algebra` as Johan says), but I think we should keep unicode in notations with ascii for identifiers (possibly with exception for foreign names).